### DISCUSSION TITLE
Eager loading slugs?

### DISCUSSION THREAD

**Discussion by BezPowell** [area17/twill#2209] [Q&A] [answered]
Apologies if this is answered elsewhere. I'm fairly new to laravel as well as twill and couldn't find an answer in the documentation.

I have a model called page on a test site and I'm loading a collection of them on every page to use in the menu. They have slugs and I've discovered by using Debug Bar that each related slug is being fetched with a separate query.

I tried adding the 'page_slugs' database table to the view with `with()`, but get an error `Call to undefined relationship [page_slugs] on model [App\Models\Page].`. It is possible to eager-load slugs and, if so, how would I go about doing so?

My code in `AppServiceProvider` is:
```
View::composer('includes.nav', function ($view) {
          $pages = Page::with('descendants', 'page_slugs')->where('published', 1)
              ->where('include_in_menu', 1)
              ->orderBy('position')
              ->get();

          // Move descendant pages into sub-list
          $sortedPages = [];
          foreach ($pages as $item) {
              if (is_null($item['parent_id'])) {
                  $item['descendants'] = $item->descendants;
                  $sortedPages[] = $item;
              }
          }

          $view->with('pages', $sortedPages);
      });
```
Again, I've probably missed something painfully obvious, but am just starting to get my head round how laravel lazy_loads tables.

**Comment by ifox** [comment, marked as answer]
Hi @BezPowell the relationship you are trying to eager load is named "slugs", not "page_slugs". You're on the right path by using with() on your query.

**Reply by BezPowell** [reply]
Thank you very much for such a quick response.

I tried 'slug' and 'page_slugs', but hadn't tried 'slugs'. The number of database queries is looking much more reasonable now. I'll get my head round relationships in Laravel eventually.

**Comment by ifox** [comment]
However, you don't need to load all slugs (as in inactive ones too), so you can also limit the eager loading with with() to only load active ones for the current locale. See https://laravel.com/docs/10.x/eloquent-relationships#constraining-eager-loads.
